hack/scripts/prow-ci-operator-deploy.sh

```diff
@@ -1,4 +1,205 @@
+        then
+            echo -e "\nno $NAMESPACE namespace found\n"
+            return
+	fi
+
+    $OC delete namespace $NAMESPACE || true
+}
+
+function cleanupPre {
+    cleanup    
+}
+
+function getEnvVariables {
+    if [[ -z $USE_ENVRC ]];
+        then
+            export FAKE_KEY_VALUE=$(cat /tmp/secret/aao-aws-creds/FAKE_KEY_VALUE)
+            export FORCE_DEV_MODE=cluster
+            export OSD_STAGING_1_AWS_ACCOUNT_ID=$(cat /tmp/secret/aao-aws-creds/OSD_STAGING_1_AWS_ACCOUNT_ID)
+            export OSD_STAGING_2_AWS_ACCOUNT_ID=$(cat /tmp/secret/aao-aws-creds/OSD_STAGING_2_AWS_ACCOUNT_ID)
+            export STS_ROLE_ARN=$(cat /tmp/secret/aao-aws-creds/STS_ROLE_ARN)
+            export STS_JUMP_ARN=$(cat /tmp/secret/aao-aws-creds/STS_JUMP_ARN)
+            export OSD_STAGING_1_OU_ROOT_ID=$(cat /tmp/secret/aao-aws-creds/OSD_STAGING_1_OU_ROOT_ID)
+            export OSD_STAGING_1_OU_BASE_ID=$(cat /tmp/secret/aao-aws-creds/OSD_STAGING_1_OU_BASE_ID)
+            export SUPPORT_JUMP_ROLE=$(cat /tmp/secret/aao-aws-creds/SUPPORT_JUMP_ROLE)
+            export STS_JUMP_ROLE=$(cat /tmp/secret/aao-aws-creds/STS_JUMP_ROLE)
+            export OPERATOR_ACCESS_KEY_ID=$(cat /tmp/secret/aao-aws-creds/OPERATOR_ACCESS_KEY_ID)
+            export OPERATOR_SECRET_ACCESS_KEY=$(cat /tmp/secret/aao-aws-creds/OPERATOR_SECRET_ACCESS_KEY)
+        else
+            source .envrc
+    fi
+
+    if [[ -z $OSD_STAGING_1_AWS_ACCOUNT_ID ]]; then
+        return 1
+    fi
+}
+
+function preDeploy {
+    echo -e "\nDEPLOY CRDs, Secret, Config Map\n"
+    $OC adm new-project "$NAMESPACE" || true
+    if [ -z $IS_LOCAL -a -z $IS_STAGING ];
+        then
+            make prow-ci-predeploy
+            return
+        else
+            make predeploy
+    fi
+    source .venv/bin/activate
+}
+
+function createDockerfileSoftLink {
+    ln ./build/Dockerfile Dockerfile
+}
+
+function buildOperatorImage {
+    echo -e "\nSTARTING BUILD IMAGE\n"
+    $OC_WITH_NAMESPACE new-build --binary --strategy=docker --build-arg=FIPS_ENABLED=false  --name $BUILD_CONFIG || true
+    $OC_WITH_NAMESPACE start-build $BUILD_CONFIG --from-dir . -F
+    $OC_WITH_NAMESPACE set image-lookup $BUILD_CONFIG
+}
+
+function verifyBuildSuccess {
+    sleep 5
+    local latestJobName phase
+    latestJobName="$BUILD_CONFIG"-$($OC_WITH_NAMESPACE get buildconfig "$BUILD_CONFIG" -ojsonpath='{.status.lastVersion}') 
+    phase=$($OC_WITH_NAMESPACE get build "$latestJobName" -ojsonpath='{.status.phase}')
+    if [[ $phase != "Complete" ]]; then
+        echo "build was not completed fully, the state was $phase but expected to be 'Complete'"
+        echo "the logs for the failed job are:"
+
+        $OC_WITH_NAMESPACE logs $latestJobName-build
+        return 1
+    fi
+}
+
+function deployOperator {
+  echo -e "\nDEPLOYING OPERATOR\n"
+
+  make prow-ci-deploy-image OPERATOR_IMAGE_URI=$BUILD_CONFIG
+}
+
+function waitForDeployment {
```

Comment:
Might be worth to check if just using `kubectl wait` works here - might make the code a lot shorter.

Reply:
`wait` works well as a replacement. Please review the below modified code, please also find the logs for successful run here:-  https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_aws-account-operator/708/pull-ci-openshift-aws-account-operator-master-integration-test/1551988141563318272/artifacts/integration-test/test/build-log.txt